sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridClient.cs

```diff
@@ -74,7 +75,7 @@ public EventGridClient(Uri endpoint, AzureKeyCredential credential, EventGridCli
         public EventGridClient(Uri endpoint, SharedAccessSignatureCredential credential, EventGridClientOptions options)
         {
             Argument.AssertNotNull(credential, nameof(credential));
-            options ??= new EventGridClientOptions();
+            _clientOptions = options ?? new EventGridClientOptions();
```

Comment:
Typically, we store off the thing that we need from options rather than the whole options bag. This helps keep things more immutable as it protects us against the user changing the serializer on their options reference after already creating the client. If the serializer is null we can set our field to new JsonObjectSerializer here.